### PR TITLE
editorconfig

### PR DESCRIPTION
**Pull request by seif**
- Add rider .idea to gitignore
- Add gitattributes file
- Add editorconfig
- Dont require this qualifier
- Prefix private fields with undescores

### CODE REVIEW

**Comment by seif** [comment]
Have a bunch of warning that need to be handled after adding the editorconfig.

**Comment by sgerrand** [comment]
I'll defer to the @duffelhq/dotnet-maintainers gurus on this one!

**Review by sgerrand — APPROVED** [review]
LGTM. 👍

**Pull request merged**
